controllers/provider-aws/pkg/controller/worker/machine_controller_manager.go

```diff
@@ -0,0 +1,129 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/imagevector"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	"github.com/gardener/gardener-extensions/pkg/controller/worker"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var mcmChartPath = filepath.Join(aws.InternalChartsPath, aws.MachineControllerManagerName)
+
+func (w *workerDelegate) DeployMachineControllerManager(ctx context.Context) (string, error) {
+	mcmImage, err := imagevector.ImageVector().FindImage(aws.MachineControllerManagerImageName, w.serverVersion, w.shootVersion.GitVersion)
+	if err != nil {
+		return "", err
+	}
+
+	mcmSecretData, err := worker.GetMachineControllerManagerSecretData(ctx, w.client, w.worker.Namespace, aws.MachineControllerManagerName)
+	if err != nil {
+		return "", err
+	}
+
+	namespace := &corev1.Namespace{}
+	if err := w.client.Get(ctx, kutil.Key(w.worker.Namespace), namespace); err != nil {
+		return "", err
+	}
+
+	var (
+		seedValues = map[string]interface{}{
+			"images": map[string]interface{}{
+				aws.MachineControllerManagerName: mcmImage.String(),
+			},
+			"providerName": aws.Name,
+			"secret": map[string]interface{}{
+				"data": mcmSecretData,
+			},
+			"namespace": map[string]interface{}{
+				"uid": namespace.UID,
+			},
+		}
+
+		shootValues = map[string]interface{}{
+			"providerName": aws.Name,
+		}
+	)
+
+	// If the shoot is hibernated then we want to scale down the machine-controller-manager. However, we want to first allow it to delete
+	// all remaining worker nodes. Hence, we cannot set the replicas=0 here (otherwise it would be offline and not able to delete the nodes).
+	if extensionscontroller.IsHibernated(w.cluster.Shoot) {
+		deployment := &appsv1.Deployment{}
+		if err := w.client.Get(ctx, kutil.Key(w.worker.Namespace, aws.MachineControllerManagerName), deployment); err != nil && !apierrors.IsNotFound(err) {
+			return "", err
+		}
+		if replicas := deployment.Spec.Replicas; replicas != nil {
+			seedValues["replicas"] = *replicas
+		}
+	}
+
+	if err := w.seedChartApplier.ApplyChart(ctx, filepath.Join(mcmChartPath, "seed"), w.worker.Namespace, aws.MachineControllerManagerName, seedValues, nil); err != nil {
```

Comment:
Perhaps you could use `github.com/gardener/gardener/pkg/utils/chart` similarly to the way it's used in controlplane controllers? You could either do this in the `workerDelegate` code, or in the generic `Actuator` code (thus generalizing the flow and turning the worker delegates into "value providers" for the MCM charts).